src/core/meshcop/commissioner.cpp

```diff
@@ -418,22 +424,42 @@ ThreadError Commissioner::SendMgmtCommissionerSetRequest(const otCommissioningDa
     memset(&messageInfo, 0, sizeof(messageInfo));
     mNetif.GetMle().GetLeaderAloc(messageInfo.GetPeerAddr());
     messageInfo.mPeerPort = kCoapUdpPort;
-    SuccessOrExit(error = mSocket.SendTo(*message, messageInfo));
+    SuccessOrExit(error = mCoapClient.SendMessage(*message, messageInfo,
+                                                  Commissioner::HandleMgmtCommissionerSetResponse, this));
 
     otLogInfoMeshCoP("sent MGMT_COMMISSIONER_SET.req to leader");
 
 exit:
 
     if (error != kThreadError_None && message != NULL)
     {
-        mIsSendMgmtCommRequest = false;
         message->Free();
     }
 
     otLogFuncExitErr(error);
     return error;
 }
 
+void Commissioner::HandleMgmtCommissionerSetResponse(void *aContext, otCoapHeader *aHeader, otMessage aMessage,
+                                                     ThreadError aResult)
+{
+    static_cast<Commissioner *>(aContext)->HandleMgmtCommissisonerSetResponse(
+        static_cast<Coap::Header *>(aHeader), static_cast<Message *>(aMessage), aResult);
+}
+
+void Commissioner::HandleMgmtCommissisonerSetResponse(Coap::Header *aHeader, Message *aMessage, ThreadError aResult)
+{
+    (void) aMessage;
+
+    otLogFuncEntry();
```

Comment:
If you add a `otLogFuncEntry()` you should also have a `otLogFuncExit()` at the end of the function.